tests/systemtests/upgrade_test.go

```diff
@@ -0,0 +1,129 @@
+//go:build system_test
+
+package systemtests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+
+	systest "cosmossdk.io/systemtests"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+)
+
+const (
+	testSeed            = "scene learn remember glide apple expand quality spawn property shoe lamp carry upset blossom draft reject aim file trash miss script joy only measure"
+	upgradeHeight int64 = 22
+	upgradeName         = "v050-to-v053" // must match UpgradeName in simapp/upgrades.go
+)
+
+type initAccount struct {
+	address string
+	balance string
+}
+
+func createLegacyBinary(t *testing.T, extraAccounts ...initAccount) (*systest.CLIWrapper, *systest.SystemUnderTest) {
+	t.Helper()
+
+	legacyBinary := systest.WorkDir + "/binaries/v0.50/simd"
+
+	//// Now we're going to switch to a v.50 chain.
+	t.Logf("+++ legacy binary: %s\n", legacyBinary)
+
+	// setup the v50 chain. v53 made some changes to testnet command, so we'll have to adjust here.
+	// this only uses 1 node.
+	legacySut := systest.NewSystemUnderTest("simd", systest.Verbose, 1, 1*time.Second)
+	// we need to explicitly set this here as the constructor infers the exec binary is in the "binaries" directory.
+	legacySut.SetExecBinary(legacyBinary)
+	legacySut.SetTestnetInitializer(systest.LegacyInitializerWithBinary(legacyBinary, legacySut))
+	legacySut.SetupChain()
+	v50CLI := systest.NewCLIWrapper(t, legacySut, systest.Verbose)
+	v50CLI.AddKeyFromSeed("account1", testSeed)
+
+	modifications := [][]string{
+		{"genesis", "add-genesis-account", v50CLI.AddKey("foo"), "10000000000stake"},
+		{"genesis", "add-genesis-account", v50CLI.AddKey("bar"), "10000000000stake"},
+		{"genesis", "add-genesis-account", v50CLI.AddKey("baz"), "10000000000stake"},
+	}
```

Comment:
lets add a comment here:

"typically, SystemUnderTest will create a node with 4 validators. In the legacy setup, we create run a single validator network. This means we need to add 3 more accounts in order to make further account additions map to the same account number in state."